admin/sql/updates/20200630-mbs-10921-unlist-unrelated-edits.sql

```diff
@@ -0,0 +1,286 @@
+        FROM recording r
+        JOIN artist_credit ac ON r.artist_credit = ac.id
+        JOIN artist_credit_name acn ON ac.id = acn.artist_credit
+        WHERE acn.artist = ea.artist
+    )
+    LIMIT 900 -- of 86841
+)
+RETURNING *;
+
+\echo MBS-10921 (3/12): Unlist “Remove relationship” edits from artists’ history...
+DELETE FROM edit_artist
+WHERE (edit, artist) IN (
+    SELECT ea.edit, ea.artist
+    FROM edit_artist ea
+    JOIN edit e ON ea.edit = e.id
+    JOIN edit_data ed ON e.id = ed.edit
+    WHERE e.id >= 70658947 AND e.id < 71006471
+    AND e.type = 92
+    AND jsonb_extract_path_text(ed.data, 'relationship', 'link', 'type', 'entity0_type') = 'recording'
+    AND jsonb_extract_path_text(ed.data, 'relationship', 'link', 'type', 'entity1_type') = 'work'
+    AND jsonb_extract_path_text(ed.data, 'relationship', 'entity0', 'id')::int IN (
+        SELECT id FROM recording
+    )
+    AND jsonb_extract_path_text(ed.data, 'relationship', 'entity0', 'id')::int NOT IN (
+        SELECT r.id
+        FROM recording r
+        JOIN artist_credit ac ON r.artist_credit = ac.id
+        JOIN artist_credit_name acn ON ac.id = acn.artist_credit
+        WHERE acn.artist = ea.artist
+    )
+    LIMIT 50 -- of 1799
+)
+RETURNING *;
+
+-- II) Recordings
+
+\echo MBS-10921 (4/12): Unlist “Reorder relationships” edits from artists’ history...
+DELETE FROM edit_recording
+WHERE (edit, recording) IN (
+    SELECT ea.edit, ea.artist
+    FROM edit_artist ea
+    JOIN edit e ON ea.edit = e.id
+    JOIN edit_data ed ON e.id = ed.edit
+    WHERE e.id >= 70658947 AND e.id < 71006471
+    AND e.type = 99
+    AND jsonb_extract_path_text(ed.data, 'link_type', 'entity0_type') = 'recording'
+    AND jsonb_extract_path_text(ed.data, 'link_type', 'entity1_type') = 'work'
+    AND jsonb_extract_path_text(data, 'relationship_order', '0', 'relationship', 'entity1', 'id')::int IN (
+        SELECT id FROM recording
+    )
+    AND jsonb_extract_path_text(data, 'relationship_order', '0', 'relationship', 'entity1', 'id')::int NOT IN (
+        SELECT r.id
+        FROM recording r
+        JOIN artist_credit ac ON r.artist_credit = ac.id
+        JOIN artist_credit_name acn ON ac.id = acn.artist_credit
+        WHERE acn.artist = ea.artist
+    )
+    LIMIT 50 -- of 534
+)
+RETURNING *;
+
+\echo MBS-10921 (5/12): Unlist “Add relationship” edits from recordings’ history...
+DELETE FROM edit_recording
+WHERE (edit, recording) IN (
+    SELECT er.edit, er.recording
+    FROM edit_recording er
+    JOIN edit e ON er.edit = e.id
+    JOIN edit_data ed ON er.edit = ed.edit
+    WHERE er.edit >= 70658947 AND er.edit < 71006471
+    AND e.type = 90
+    AND jsonb_extract_path_text(ed.data, 'type0') = 'recording'
+    AND jsonb_extract_path_text(ed.data, 'type1') = 'work'
+    AND jsonb_extract_path_text(ed.data, 'entity0', 'id')::int != er.recording
+    AND jsonb_extract_path_text(ed.data, 'entity0', 'id')::int IN (
+        SELECT id FROM recording
+    )
```

Comment:
I'm not sure it matters that the linked recording still exists in this case, since if the ID is different we know it's wrong in either case?

Reply:
It doesn’t seem to be necessarily wrong. I supposed it might also be the recording id has just changed after a merge. I didn’t check if that is actually possible yet.

Reply:
Yeah, merged recordings won't be handled by any of the queries at present. You'd have to find the new ID via the `recording_gid_redirect` table. Of course I'm not sure that's worth the added effort or slowdown.